### PR TITLE
Fix for #78 #79 and #80 with some refactoring

### PR DESCRIPTION
**Pull request by guitarlum**
Hello There, 

we want to use this client in production and discovered some bugs in the area of dependency updates (and a minor one with the out io.writer). 
This pull request addresses those issues. 

Would be happy to have this merged, as we have to go with shell based workarounds in the meantime.

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -792,6 +760,37 @@ func (c *HelmClient) rollbackRelease(spec *ChartSpec, version int) error {
...
+					Getters:          c.Providers,
+					RepositoryConfig: c.Settings.RepositoryConfig,
+					RepositoryCache:  c.Settings.RepositoryCache,
+					Out:              c.output,
+				}
+				if err := man.Update(); err != nil {
+					return nil, err
+				}
+
+				helmChart, chartPath, err = c.getChart(spec.ChartName, chartPathOptions)
```
Comment (elenz97) [inline client.go:781]: `chartPath` appears to be unused. As it is not returned anyway, we can ignore it here:
```suggestion
				helmChart, _, err = c.getChart(spec.ChartName, chartPathOptions)
```

**Review by elenz97 — CHANGES_REQUESTED** [review]

File: client.go (hunk 1 above)
Comment (guitarlum) [reply client.go]: Done, thanks for the feedback @elenz97 .

**Review by elenz97 — APPROVED** [review]

**Comment by elenz97** [comment]
lgtm 👍

**Comment by guitarlum** [comment]
Thanks a lot @elenz97 . 

In general: how often / when are releases built?

**Comment by elenz97** [comment]
> In general: how often / when are releases built?

Generally speaking, we have no fixed release schedule i.e. for `go-helm-client`. I've published [v0.9.2](https://github.com/mittwald/go-helm-client/releases/tag/v0.9.2) just now, thanks for your contributions!

**Pull request merged**